apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -375,20 +375,30 @@ const OutOfOfficePage = () => {
   }, [openModalOnStart]);
 
   const [openModal, setOpenModal] = useState(false);
+
+  const { data } = trpc.viewer.outOfOfficeReasonList.useQuery();
+  // TODO: The best solution is we should find the cause of full page flicker
+  // Taling about hydration problem, migrating to appDir might solve the problem automatically
+  // which will reset the Modal state, which makes bad UX and fails (superhuman click speed) E2E tests
+  const isBeforeFlicker = data?.length === undefined || data.length <= 0;
   return (
     <>
       <Meta
         title={t("out_of_office")}
         description={t("out_of_office_description")}
         borderInShellHeader={false}
         CTA={
-          <Button
-            color="primary"
-            className="flex w-20 items-center justify-between px-4"
-            onClick={() => setOpenModal(true)}
-            data-testid="add_entry_ooo">
-            <Icon name="plus" size={16} /> {t("add")}
-          </Button>
+          isBeforeFlicker ? (
+            <SkeletonText className="h-8 w-20" />
+          ) : (
+            <Button
+              color="primary"
+              className="flex w-20 items-center justify-between px-4"
+              onClick={() => setOpenModal(true)}
+              data-testid="add_entry_ooo">
+              <Icon name="plus" size={16} /> {t("add")}
+            </Button>
+          )
```

Comment:
Nah, this does not look good for rendering the button conditionally. Instead, you should disable the button

Reply:
Because it's non-interactive due to waiting for several data to be loaded, isn't the loading state better than just the disabled state? LMKWYT.